hack/manifest-cleaner/yaml.go

```diff
@@ -123,3 +125,95 @@ func nullDocument(n *yaml.Node) bool {
 	}
 	return len(n.Content) == 0 || n.Content[0].Tag == "!!null"
 }
+
+const rbacCheckTag = "rbac-check"
+
+// filterOutRBACCheckTags returns the given list of YAML nodes with all nested
+// rbac-check tags filtered out.
+func filterOutRBACCheckTags(docs []*yaml.Node) []*yaml.Node {
+	it := nodesWithRBACCheckTag(docs...)
+
+	for node, ok := it(); ok; node, ok = it() {
+		*node = *removeRBACCHeckTags(node)
+	}
+
+	return docs
+}
```

Comment:
The high-level logic is here.

1. Craft a YAML iterator that returns _only_ the nodes that have a `+rbac-check` tag inside their comment, such as
    ```yaml
    # Something
    # +rbac-check:subresource=something
    - apiGroups:
      - sources.triggermesh.io
    ```
1. For each of these nodes, trim lines which are identified as `+rbac-check` tag:
    ```yaml
    # Something
    - apiGroups:
      - sources.triggermesh.io
    ```